mlir-tensorrt/compiler/lib/Compiler/TensorRTToExecutable/Passes.cpp

```diff
@@ -124,6 +124,42 @@ outlineOp(RewriterBase &rewriter, tensorrt::TensorRTModuleOp trtModule,
   if (failed(func))
     return failure();
 
+  StringRef tensorrtShapeBoundsAttrName =
+      mlir::tensorrt::TensorRTDialect::getShapeProfileArgAttrName();
+  func::FuncOp funcContainingCluster =
+      cluster.back()->getParentOfType<func::FuncOp>();
+  SmallVector<Attribute> profileAttrsPerInput;
+  for (Value v : inputs) {
+    auto rtt = dyn_cast<RankedTensorType>(v.getType());
+    if (!rtt || rtt.hasStaticShape()) {
+      profileAttrsPerInput.push_back(Attribute{});
+      continue;
+    }
+
+    auto blockArg = dyn_cast<BlockArgument>(v);
+    if (!blockArg ||
+        blockArg.getOwner()->getParentOp() != funcContainingCluster)
+      emitError(trtModule.getLoc())
+          << "Block argument is not part of the signature of the function "
+             "containing this TRT cluster";
+
+    int64_t argIndex = blockArg.getArgNumber();
+    profileAttrsPerInput.push_back(
+        funcContainingCluster.getArgAttrOfType<tensorrt::ShapeProfileAttr>(
+            argIndex, tensorrtShapeBoundsAttrName));
+
+    if (!profileAttrsPerInput.back())
+      emitError(trtModule.getLoc())
```

Comment:
You're emitting an error here but not communicating that the pass should fail (if an error is emitted, then the pass should fail).

You can do `return emitError(...);` which is equivalent to  `emitError(...); return failure();`. 

Then the failure needs to be passed backwards to the `runOnOperation` function in the pass, where you can then do `signalPassFailure();`

Reply:
Might also want to give a more descriptive error message (these are visible to compiler user). For example, include the key of the attribute that is expected (`tensorrtShapeBoundsAttrName`)